kubernetes/heapster.go

```diff
@@ -114,221 +111,26 @@ func getHeapsterURL(clientset kubernetes.Interface, clusterHostURL string) (URL
 
 }
 
-//ensure Heapster is accessible or launch a copy
-func ensureValidHeapster(config KubeAgentConfig) (KubeAgentConfig, error) {
-
-	client := &config.HTTPClient
-	var err error
-
-	// if Heapster is found in the cluster, test to make sure we can connect to it
-	// or if not using node summaries and unable to connect or find heapster in the kube-system namespace
-	// launch an instance of heapster in the cloudability namespace. Make sure and close the returned response body.
-	if config.HeapsterURL != "" {
-		return validateHeapster(config, client, config.Namespace)
-	}
-	if !config.RetrieveNodeSummaries {
-		log.Printf("Could not find heapster running in the cluster, launching a copy in the %s namespace.", config.Namespace)
-
-		config.HeapsterURL, err = launchHeapster(config.Clientset, config.UseInClusterConfig, config.Namespace)
-
-		if err != nil {
-			return config, fmt.Errorf("Error launching heapster in the %s namespace: %+v", err, config.Namespace)
-		}
-		innerTest, _, err := util.TestHTTPConnection(
-			client, config.HeapsterURL, http.MethodGet, config.BearerToken, retryCount, true)
-		if innerTest || err == nil {
-			log.Printf("Connected to heapster at: %v", config.HeapsterURL)
-		}
-	}
-
-	return config, err
-}
-
-func validateHeapster(config KubeAgentConfig, client rest.HTTPClient, namespace string) (KubeAgentConfig, error) {
+func validateHeapster(config KubeAgentConfig, client rest.HTTPClient) error {
 	outerTest, body, err := util.TestHTTPConnection(
 		client, config.HeapsterURL, http.MethodGet, config.BearerToken, retryCount, true)
 
-	var me heapsterMetricExport
+	if err != nil {
+		return err
+	}
 
+	var me heapsterMetricExport
 	if err := json.Unmarshal(*body, &me); err != nil {
 		log.Printf("malformed response from heapster running at: %v", config.HeapsterURL)
-	}
-
-	if !outerTest {
-		log.Printf("Could not connect to heapster at: %v, launching a copy in the Cloudability namespace.",
+	} else if !outerTest {
+		return fmt.Errorf("no heapster found")
```

Comment:
seems like this block should go first. and may be able to take out the if/else structures (if the ifs are returning...)